src/qopenglwebpage.cpp

```diff
@@ -175,6 +176,7 @@ void QOpenGLWebPage::drawUnderlay()
 */
 void QOpenGLWebPage::drawOverlay(const QRect &rect)
 {
+    mGrabResultListLock.lock();
```

Comment:
Why not QMutexLocker over here? I'd have a scope between 179-191 (excluding afterRendering signal) and use QMutexLocker lock(&mGrabResultListLock) within the scope even though you are not returning between 179 and 190.